R/carehomes.R

```diff
@@ -851,25 +1021,100 @@ carehomes_compare <- function(state, observed, pars) {
                             observed$sero_tot_15_64_2,
                             model_sero_prob_pos_2)
 
+  ## Stop if both age-specific and aggregated pillar 2 data will be passed to
+  ## log-likelihood calculation!
+  if (any(!is.na(observed$pillar2_over25_tot)) &&
+      any(!is.na(observed$pillar2_25_49_tot)) &&
+      all(is.na(observed$pillar2_over25_cases)) &&
+      all(is.na(observed$pillar2_25_49_cases))) {
+    stop("Cannot double fit age-specific and aggregated pillar 2")
+  }
+  if (any(!is.na(observed$pillar2_over25_cases)) &&
+      any(!is.na(observed$pillar2_25_49_cases)) &&
+      all(is.na(observed$pillar2_over25_tot)) &&
+      all(is.na(observed$pillar2_25_49_tot))) {
+    stop("Cannot double fit age-specific and aggregated pillar 2")
+  }
```

Comment:
These checks should be in the `carehomes_particle_filter_data` function - ideally they should be comprehensive (currently these checks would allow us to double fit e.g. over25 and 50_64)

Reply:
done